Documentation/design-docs/framework-version-resolution.md

```diff
@@ -314,3 +317,38 @@ There should not be any circular dependencies between frameworks.
 A newer version of a shared framework should keep or increase the version to another shared framework (never decrease the version number).
 
 By following these best practices we have optimal run-time performance (less processing and probing) and less chance of incompatible framework references.
+
+
+## Changes to existing apps
+The above proposal will impact behavior of existing apps (because framework resolution is in `hostfxr` which is global on the machine for all frameworks). This is a description of the changes as they apply to apps using either default, `rollForwardOnNoCandidateFx` or `applyPatches`.
+
+### Roll on patches-only will now roll from release to pre-release if no release is available
+When `rollForwardOnNoCandidateFx` is disabled (set to `0` which is not the default) the existing behavior is to never roll forward to a pre-release version. If the settings is any other value (Minor/Major) it would roll forward to pre-release version if there's no available matching release version.
+
+So for example, the machine has only `3.0.1-preview.1` installed and the application has reference to `3.0.0`. The existing behavior is
+* Default behavior is to roll forward to the `3.0.1-preview.1` since there's no matching release version, and run the app.
+* If the `rollForwardOnNoCandidateFx=0` (and only then), the app will fail to run (as it won't roll forward to pre-release version).
+
+The new behavior will be to treat all settings of `rollForwardOnNoCandidateFx` the same with regard to pre-release. That is release version will roll forward to pre-release if there's no release version available. In the above sample, the app would run using the `3.0.1-preview.1` framework.
+
+| Reference                                     | Available versions | Existing behavior | New behavior      |
+| --------------------------------------------- | ------------------ | ----------------- | ----------------- |
+| 3.0.0                                         | 3.0.1-preview.1    | 3.0.1-preview.1   | 3.0.1-preview.1   |
+| 3.0.0 rollForwardOnNoCandidateFx=0            | 3.0.1-preview.1    | failure           | 3.0.1-preview.1   |
+
+
+### Pre-release will roll forward
+The existing behavior is that pre-release only rolls forward to the closest higher pre-release of the same `major.minor.patch`. This also means that if there's an exact match available, pre-release doesn't roll forward. Pre-release also never rolls forward to release.
+
+With the proposed behavior pre-release will be treated the same way as release, that is roll forward on patches, minor or major based on the settings. The only difference that it will always consider both release and pre-release versions (unlike release which prefers to roll forward to release).  
```

Comment:
```suggestion
With the proposed behavior pre-release will be treated the same way as release: roll forward on patches, minor or major based on the settings. The only difference is that it will always consider both release and pre-release versions (unlike release which prefers to roll forward to release).  
```